app/audit/models/TaxCalculationDetailsResponseAuditModel.scala

```diff
@@ -0,0 +1,315 @@
+
+
+  private def deductionsString(deduction: String): String =
+    deduction match {
+      case "inYearAdjustment" => "Outstanding debt collected through PAYE"
+      case "payeEmployments" => "All employments"
+      case "ukPensions" => "UK pensions"
+      case "stateBenefits" => "State benefits"
+      case "cis" => "CIS and trading income"
+      case "ukLandAndProperty" => "UK land and property"
+      case "specialWithholdingTax" => "Special withholding tax"
+      case "voidISAs" => "Void ISAs"
+      case "savings" => "Interest received from UK banks and building societies"
+      case _ => deduction
+    }
+
+  private def taxDeductionsJson(taxDeductions: (String, BigDecimal)): JsObject = Json.obj(
+    "deductionsType" -> deductionsString(taxDeductions._1),
+    "amount" -> taxDeductions._2
+  )
+
+  private val taxDeductionsFullSeq: Seq[(String, BigDecimal)] = {
+    val totalDeductions = calculation.taxDeductedAtSource.total
+    totalDeductions.fold(calculation.taxDeductedAtSource.allFields)(total => calculation.taxDeductedAtSource.allFields :+ ("Income Tax due after deductions", total))
+  }
+
+  private def optDetail(optDetail: Seq[JsObject]): Option[Seq[JsObject]] =
+    if(optDetail.nonEmpty) Some(optDetail) else None
+
+  private def optDetail(optDetail: JsObject): Option[JsObject] = {
+    if(optDetail.equals(Json.obj())) None else Some(optDetail)
+  }
+
+
+  private val calculationMessagesDetail: Option[Seq[JsObject]] = optDetail(allowedCalcMessages.map(calcMessagesJson))
+
+  private val payPensionsProfitDetail: Option[Seq[JsObject]] = optDetail(calculation.payPensionsProfit.bands.map(taxBandRateMessageJson))
+
+  private val savingsDetail: Option[Seq[JsObject]] = optDetail(calculation.savingsAndGains.bands.map(taxBandRateMessageJson))
+
+  private val dividendsDetail: Option[Seq[JsObject]] = optDetail(calculation.dividends.bands.map(taxBandRateMessageJson))
+
+  private val employmentLumpSumsDetail: Option[Seq[JsObject]] = optDetail(calculation.lumpSums.bands.map(taxBandRateMessageJson))
+
+  private val gainsOnLifePoliciesDetail: Option[Seq[JsObject]] = optDetail(calculation.gainsOnLifePolicies.bands.map(taxBandRateMessageJson))
+
+  private val class4NationInsuranceDetail: Option[Seq[JsObject]] = optDetail(calculation.nic.class4Bands.getOrElse(Seq.empty).map(nicBandRateMessageJson))
+
+  private val taxReductionsDetails: Option[Seq[JsObject]] = optDetail(taxReductionsFullSeq.map(taxReductionsJson))
+
+  private val additionChargesDetail: Option[Seq[JsObject]] = optDetail(additionalChargesSeq.map(lineItemJson))
+
+  private val capitalGainsTaxDetail: Option[JsObject] = optDetail(capitalGainsTaxJson(calculation.capitalGainsTax))
+
+  private val otherChargesDetail: Option[Seq[JsObject]] = optDetail(otherChargesSeq.map(lineItemJson))
+
+  private val taxDeductionsDetail: Option[Seq[JsObject]] = optDetail(taxDeductionsFullSeq.map(taxDeductionsJson))
+
+  override val detail: JsValue =
```

Comment:
Worth to raise with BAs that we don't audit the field `incomeTaxDueAfterOtherCharges` and don't show it on the actual page either.